src/main/java/com/google/devtools/build/lib/rules/repository/NewRepositoryFileHandler.java

```diff
@@ -167,75 +158,32 @@ public void finishFile(Rule rule, Path outputDirectory, Map<String, String> mark
       }
     }
 
-    private String getFileAttributeValue(Rule rule) throws RepositoryFunctionException {
-      WorkspaceAttributeMapper mapper = WorkspaceAttributeMapper.of(rule);
-      String fileAttribute;
+    private Label getFileAttributeAsLabel(Rule rule) throws RepositoryFunctionException {
       try {
-        fileAttribute = mapper.get(getFileAttrName(), Type.STRING);
+        return WorkspaceAttributeMapper.of(rule).get(getFileAttrName(), BuildType.NODEP_LABEL);
       } catch (EvalException e) {
         throw new RepositoryFunctionException(e, Transience.PERSISTENT);
       }
-      return fileAttribute;
-    }
-
-    private Label getFileAttributeAsLabel(Rule rule) throws RepositoryFunctionException {
-      Label label;
-      try {
-        // Parse a label
-        label = Label.parseCanonical(getFileAttributeValue(rule));
-      } catch (LabelSyntaxException ex) {
-        throw new RepositoryFunctionException(
-            Starlark.errorf(
-                "the '%s' attribute does not specify a valid label: %s",
-                getFileAttrName(), ex.getMessage()),
-            Transience.PERSISTENT);
-      }
-      return label;
     }
 
     @Nullable
     private FileValue getFileValue(Rule rule, Environment env)
         throws RepositoryFunctionException, InterruptedException {
-      String fileAttribute = getFileAttributeValue(rule);
-      RootedPath rootedFile;
-
-      if (LabelValidator.isAbsolute(fileAttribute)) {
-        Label label = getFileAttributeAsLabel(rule);
-        SkyKey pkgSkyKey = PackageLookupValue.key(label.getPackageIdentifier());
-        PackageLookupValue pkgLookupValue = (PackageLookupValue) env.getValue(pkgSkyKey);
-        if (pkgLookupValue == null) {
-          return null;
-        }
-        if (!pkgLookupValue.packageExists()) {
-          throw new RepositoryFunctionException(
-              Starlark.errorf("Unable to load package for %s: not found.", fileAttribute),
-              Transience.PERSISTENT);
-        }
-
-        // And now for the file
-        Root packageRoot = pkgLookupValue.getRoot();
-        rootedFile = RootedPath.toRootedPath(packageRoot, label.toPathFragment());
-      } else {
-        // TODO(dmarting): deprecate using a path for the workspace_file attribute.
-        PathFragment file = PathFragment.create(fileAttribute);
-        Path fileTarget = workspacePath.getRelative(file);
-        if (!fileTarget.exists()) {
-          throw new RepositoryFunctionException(
-              Starlark.errorf(
-                  "the '%s' attribute does not specify an existing file (%s does not exist)",
-                  getFileAttrName(), fileTarget),
-              Transience.PERSISTENT);
-        }
-
-        if (file.isAbsolute()) {
```

Comment:
Do I understand correctly that this change removes the ability of `new_local_repository` to use BUILD files from an absolute path?

(I think it's a good change per se and that use case looks weird enough that I'd be fine with not having an incompatible flag as long as you do it knowingly)